File: src/main/java/boardgame/com/constant/PlayStatus.java
package boardgame.com.constant;

import boardgame.com.exception.ApiException;
import boardgame.com.exception.ApiExceptionEnum;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.Arrays;

@Getter
@AllArgsConstructor
@JsonFormat(shape = JsonFormat.Shape.OBJECT)
public enum PlayStatus {

    STAND_BY("C002", "4","대기", 1, ""),
    PLAYING("C002", "1","진행중", 2, ""),
    FINISHED("C002", "2","종료", 3, ""),
    ABNORMAL("C002", "3","비정상", 4, ""),
    ;

    private final String groupCode;
    private final String code;
    private final String title;
    private final long order;
    private final String remark;

    public static PlayStatus ofCode(String code) {
        return Arrays.stream(PlayStatus.values())
                .filter(value -> value.getCode().equals(code))
                .findAny()
                .orElseThrow(() -> {
                    throw new ApiException(ApiExceptionEnum.CODE_NOT_EXISTS);
                });
    }
    
}

File: src/main/webapp/js/com/playStatus.js
class PlayStatus {

    static STAND_BY = {
        code: "4",
        name: "STAND_BY",
        title: "대기"
    };

    static PLAYING = {
        code: "1",
        name: "PLAYING",
        title: "진행중"
    };

    static FINISHED = {
        code: "2",
        name: "FINISHED",
        title: "종료",
    };

    static ABNORMAL = {
        code: "3",
        name: "ABNORMAL",
        title: "비정상",
    };

    static ofCode = code => {
        if (this.STAND_BY.code == code) return this.STAND_BY;
        if (this.PLAYING.code == code) return this.PLAYING;
        if (this.FINISHED.code == code) return this.FINISHED;
        if (this.ABNORMAL.code == code) return this.ABNORMAL;
    }

}
